x/mongo/driver/topology/CMAP_spec_test.go

```diff
@@ -210,76 +288,65 @@ func checkEvents(t *testing.T, expectedEvents []cmapEvent, actualEvents chan *ev
 				}
 			}
 		}
 
 		if expectedEvent.ConnectionID != 0 {
 			if expectedEvent.ConnectionID == 42 {
 				if validEvent.ConnectionID == 0 {
-					t.Fatalf("expected a connectionId but found none in %v", validEvent.Type)
+					t.Errorf("expected a connectionId but found none in %v", validEvent.Type)
 				}
 			} else if expectedEvent.ConnectionID != validEvent.ConnectionID {
-				t.Fatalf("expected and actual connectionIds differed: expected: %v, actual: %v for event: %v", expectedEvent.ConnectionID, validEvent.ConnectionID, expectedEvent.EventType)
+				t.Errorf("expected and actual connectionIds differed: expected: %v, actual: %v for event: %v", expectedEvent.ConnectionID, validEvent.ConnectionID, expectedEvent.EventType)
 			}
 		}
 
 		if expectedEvent.Reason != "" && expectedEvent.Reason != validEvent.Reason {
-			t.Fatalf("event reason differed from expected: expected: %v, actual: %v for %v", expectedEvent.Reason, validEvent.Reason, expectedEvent.EventType)
+			t.Errorf("event reason differed from expected: expected: %v, actual: %v for %v", expectedEvent.Reason, validEvent.Reason, expectedEvent.EventType)
 		}
 
 		if expectedEvent.Options != nil {
 			if expectedEvent.Options == float64(42) {
 				if validEvent.PoolOptions == nil {
-					t.Fatalf("expected poolevent options but found none")
+					t.Errorf("expected poolevent options but found none")
 				}
 			} else {
 				opts, ok := expectedEvent.Options.(map[string]interface{})
 				if !ok {
-					t.Fatalf("event options were unexpected type: %T for %v", expectedEvent.Options, expectedEvent.EventType)
+					t.Errorf("event options were unexpected type: %T for %v", expectedEvent.Options, expectedEvent.EventType)
 				}
 
 				if maxSize, ok := opts["maxPoolSize"]; ok && validEvent.PoolOptions.MaxPoolSize != uint64(maxSize.(float64)) {
-					t.Fatalf("event's max pool size differed from expected: %v, actual: %v", maxSize, validEvent.PoolOptions.MaxPoolSize)
+					t.Errorf("event's max pool size differed from expected: %v, actual: %v", maxSize, validEvent.PoolOptions.MaxPoolSize)
 				}
 
 				if minSize, ok := opts["minPoolSize"]; ok && validEvent.PoolOptions.MinPoolSize != uint64(minSize.(float64)) {
-					t.Fatalf("event's min pool size differed from expected: %v, actual: %v", minSize, validEvent.PoolOptions.MinPoolSize)
+					t.Errorf("event's min pool size differed from expected: %v, actual: %v", minSize, validEvent.PoolOptions.MinPoolSize)
 				}
 
 				if waitQueueTimeoutMS, ok := opts["waitQueueTimeoutMS"]; ok && validEvent.PoolOptions.WaitQueueTimeoutMS != uint64(waitQueueTimeoutMS.(float64)) {
-					t.Fatalf("event's min pool size differed from expected: %v, actual: %v", waitQueueTimeoutMS, validEvent.PoolOptions.WaitQueueTimeoutMS)
+					t.Errorf("event's min pool size differed from expected: %v, actual: %v", waitQueueTimeoutMS, validEvent.PoolOptions.WaitQueueTimeoutMS)
 				}
 			}
 		}
 	}
-
-EventsLeft:
```

Comment:
Curious, is this necessary to remove?

The CMAP [test README](https://github.com/mongodb/specifications/tree/6464c0692bd0f81d92d58da0f72bf1642a703056/source/connection-monitoring-and-pooling/tests#unit-test-runner) does not prohibit extra events. So I think this is OK to remove.

Reply:
Ah OK. Then pool-checkout-maxConnecting-timeout relies on test runners ignore extra events. Updating the test runner SGTM.